base/libgit2/consts.jl

```diff
@@ -271,12 +274,14 @@ Option flags for `GitRepo`.
                         CREDTYPE_USERNAME           = Cuint(1 << 5),
                         CREDTYPE_SSH_MEMORY         = Cuint(1 << 6))
 
+if LibGit2.version() >= v"0.24.0"
```

Comment:
README specifies a minimum version of 0.21. I'm not sure how specifically this has been tested, but there are considerably newer distros than Ubuntu 14.04 out there now, that do have new enough libgit2 versions that we could use them. If supporting a range of versions is simple enough to do, it doesn't really hurt us to just do it.